<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Gene Prediction Methods | The Genome Project: Excercise Guidebook</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Gene Prediction Methods | The Genome Project: Excercise Guidebook" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Gene Prediction Methods | The Genome Project: Excercise Guidebook" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Javier F. Tabima Restrepo" />


<meta name="date" content="2023-02-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="hybrid-assembly-methods-for-short-and-long-reads.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="font-awesome.min.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The Genome Project: A Guidebook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> BIOL209 (The Genome Project) Excercise Guidebook</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#additional-information-about-the-professor"><i class="fa fa-check"></i>Additional information about the professor</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="unix-challenge.html"><a href="unix-challenge.html"><i class="fa fa-check"></i><b>2</b> UNIX Challenge</a></li>
<li class="chapter" data-level="3" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html"><i class="fa fa-check"></i><b>3</b> FASTQ files and understanding read qualities</a>
<ul>
<li class="chapter" data-level="3.1" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html#sample-distribution"><i class="fa fa-check"></i><b>3.1</b> Sample distribution</a></li>
<li class="chapter" data-level="3.2" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html#basic-information-and-creating-backups"><i class="fa fa-check"></i><b>3.2</b> Basic information and creating backups</a></li>
<li class="chapter" data-level="3.3" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html#quality-control"><i class="fa fa-check"></i><b>3.3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html#evaluating-qc-of-fastq-using-fastqc"><i class="fa fa-check"></i><b>3.3.1</b> Evaluating QC of FASTQ using FastQC</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html"><i class="fa fa-check"></i><b>4</b> Laboratory techniques for DNA extraction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#protocol-using-spin-columns"><i class="fa fa-check"></i><b>4.1</b> Protocol using spin columns</a></li>
<li class="chapter" data-level="4.2" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#quantification-of-dna-using-qubit-and-nanodrop"><i class="fa fa-check"></i><b>4.2</b> Quantification of DNA using Qubit and Nanodrop</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#qubit-flurescence-quantification"><i class="fa fa-check"></i><b>4.2.1</b> Qubit flurescence quantification</a></li>
<li class="chapter" data-level="4.2.2" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#nanodrop-dna-quality"><i class="fa fa-check"></i><b>4.2.2</b> Nanodrop DNA quality</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#analytical-gel-for-dna-integrity-and-fragment-size"><i class="fa fa-check"></i><b>4.3</b> Analytical gel for DNA integrity and fragment size</a></li>
<li class="chapter" data-level="4.4" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#dna-extraction-report"><i class="fa fa-check"></i><b>4.4</b> DNA extraction report</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html"><i class="fa fa-check"></i><b>5</b> Improving Illumina Read Quality</a>
<ul>
<li class="chapter" data-level="5.1" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html#running-trimmomatic"><i class="fa fa-check"></i><b>5.1</b> Running Trimmomatic</a></li>
<li class="chapter" data-level="5.2" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html#evaluating-quality-of-processed-reads"><i class="fa fa-check"></i><b>5.2</b> Evaluating quality of processed reads</a></li>
<li class="chapter" data-level="5.3" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html#requirements-for-processing-your-reads"><i class="fa fa-check"></i><b>5.3</b> Requirements for processing your reads</a></li>
<li class="chapter" data-level="5.4" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html#what-to-turn-in"><i class="fa fa-check"></i><b>5.4</b> What to turn in?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="nanopore-sequencing.html"><a href="nanopore-sequencing.html"><i class="fa fa-check"></i><b>6</b> Nanopore Sequencing</a>
<ul>
<li class="chapter" data-level="6.1" data-path="nanopore-sequencing.html"><a href="nanopore-sequencing.html#summarizing-our-nanopore-sequencing"><i class="fa fa-check"></i><b>6.1</b> Summarizing our Nanopore sequencing</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html"><i class="fa fa-check"></i><b>7</b> Assembly Methods for Short Read Sequencing</a>
<ul>
<li class="chapter" data-level="7.1" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#comparing-between-k-mers-in-genome-assembly"><i class="fa fa-check"></i><b>7.1</b> Comparing between k-mers in genome assembly</a></li>
<li class="chapter" data-level="7.2" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#using-a-multi-k-mer-approach-to-avoid-manual-k-mer-selection"><i class="fa fa-check"></i><b>7.2</b> Using a multi-K-mer approach to avoid manual k-mer selection</a></li>
<li class="chapter" data-level="7.3" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#cleaning-the-assembly-and-detecting-contamination"><i class="fa fa-check"></i><b>7.3</b> Cleaning the assembly and detecting contamination</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#sketch-assay"><i class="fa fa-check"></i><b>7.3.1</b> <strong>Sketch</strong> assay</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#coverage-and-taxonomy-detection-assay"><i class="fa fa-check"></i><b>7.4</b> Coverage and Taxonomy detection assay</a></li>
<li class="chapter" data-level="7.5" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#extracting-the-scaffolds-of-interest"><i class="fa fa-check"></i><b>7.5</b> Extracting the scaffolds of interest</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="hybrid-assembly-methods-for-short-and-long-reads.html"><a href="hybrid-assembly-methods-for-short-and-long-reads.html"><i class="fa fa-check"></i><b>8</b> Hybrid Assembly Methods for Short and Long Reads</a>
<ul>
<li class="chapter" data-level="8.1" data-path="hybrid-assembly-methods-for-short-and-long-reads.html"><a href="hybrid-assembly-methods-for-short-and-long-reads.html#hybrid-assembly-using-unicycler"><i class="fa fa-check"></i><b>8.1</b> Hybrid assembly using <code>Unicycler</code></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html"><i class="fa fa-check"></i><b>9</b> Gene Prediction Methods</a>
<ul>
<li class="chapter" data-level="9.1" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html#installing-prokka"><i class="fa fa-check"></i><b>9.1</b> Installing prokka</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html#installing-the-prokka-environment-and-program"><i class="fa fa-check"></i><b>9.1.1</b> Installing the prokka environment and program</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html#annotating-the-genomes-using-prokka"><i class="fa fa-check"></i><b>9.2</b> Annotating the genomes using prokka</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html#summarizing-the-results-from-prokka"><i class="fa fa-check"></i><b>9.2.1</b> Summarizing the results from prokka</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Genome Project: Excercise Guidebook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gene-prediction-methods" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Gene Prediction Methods<a href="gene-prediction-methods.html#gene-prediction-methods" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="objectives">
<p>
<em>Objectives</em>
</p>
<ol style="list-style-type: decimal">
<li>
</li>
<li>
</li>
<li>
</li>
</ol>
</div>
<p>After assembling the genome, we need to identify which regions are genes. This is called gene annotation or gene prediction, and uses a combination of <em>ab-initio</em> approaches with homology-based searches to predict and annotate the genes.</p>
<p>We will use <a href=""><code>prokka</code></a>, a state-of-the-art program for prokariotic genome annotation. This program uses both approaches to predict genes.</p>
<div id="installing-prokka" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Installing prokka<a href="gene-prediction-methods.html#installing-prokka" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>#*# Installing miniconda</p>
<p>Before we can start installing prokka, we need to create a computational environment for it to be able to run correctly. <code>Miniconda</code> provides that environment by creating a python instance that groups all the requirements for these programs to run.</p>
<div class="rmdalert">
<p>
<strong>PLEASE LET ME KNOW IF THESE STEPS DONT WORK. INSTALLING CONDA
CAN BE ANNOYING!</strong>
</p>
</div>
<pre><code>bash
/Smaug_SSD/bin/Miniconda3-latest-Linux-x86_64.sh</code></pre>
<p>Say <code>yes</code> to all the prompts, and let miniconda install.</p>
<p>Then, close your window and reconnect to Smaug.</p>
<div id="installing-the-prokka-environment-and-program" class="section level3 hasAnchor" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Installing the prokka environment and program<a href="gene-prediction-methods.html#installing-the-prokka-environment-and-program" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After logging back in, we need to install <code>Prokka</code> using miniconda as following:</p>
<pre><code>bash
conda init bash</code></pre>
<p>log out and log in again, and then:</p>
<pre><code>bash
conda create -n prokka_env -c conda-forge -c bioconda prokka
conda activate prokka_env</code></pre>
<p>We can finally run prokka. Test it by running</p>
<pre><code>prokka</code></pre>
<p>A long prompt shoud appear.</p>
<hr />
</div>
</div>
<div id="annotating-the-genomes-using-prokka" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Annotating the genomes using prokka<a href="gene-prediction-methods.html#annotating-the-genomes-using-prokka" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Usually, these computationally complex programs are very long and cumbersome to run, but the develkopers of prokka have made it super easy to run.</p>
<p>Within your sample folder, run the following code:</p>
<pre><code>prokka --genus Synechoccocus --strain the_name_of_your_strain --outdir your_sample_name_annotation --prefix your_sample_name  your_assembly.fasta</code></pre>
<p>And let it run!</p>
<div id="summarizing-the-results-from-prokka" class="section level3 hasAnchor" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> Summarizing the results from prokka<a href="gene-prediction-methods.html#summarizing-the-results-from-prokka" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After prokka is done, the outputs should be in your <code>your_sample_name_annotation</code> folder.</p>
<ol style="list-style-type: decimal">
<li>Using the <code>head</code> command, answer the following question:</li>
</ol>
<div class="rmdquestion">
<ol style="list-style-type: decimal">
<li>
Check the
<code>.fna, .fsa, .ffn, .faa, .gff, .tbl, .tsv, and .gbk</code> files
and explain what are their contents below:
</li>
</ol>
<ul>
<li>
<code>.fna</code> file:
</li>
<li>
<code>.fsa</code> file:
</li>
<li>
<code>.ffn</code> file:
</li>
<li>
<code>.faa</code> file:
</li>
<li>
<code>.gff</code> file:
</li>
<li>
<code>.tbl</code> file:
</li>
<li>
<code>.tsv</code> file:
</li>
<li>
<code>.gbk</code> file:
</li>
</ul>
<blockquote>
<p>
Hint: Check <a href="https://github.com/tseemann/prokka">the Prokka
manual</a>
</p>
</blockquote>
<ol start="2" style="list-style-type: decimal">
<li>
<p>
Using your knowledge on the command line, can you count the
number of predicted proteins, translated CDS sequences and all the
prediction transcripts (CDS, rRNA, tRNA, tmRNA, misc_RNA)
</p>
</li>
<li>
<p>
Open the <code>.txt</code> file. Answer the following
questions:
</p>
</li>
</ol>
<ul>
<li>
How many CDS are found? Do they match your counts from the previous
step?<br />
</li>
<li>
Define rRNA, tRNA and tmRNA (cite your references please) and
present the counts of each RNA type.
</li>
</ul>
<ol start="4" style="list-style-type: decimal">
<li>
Using the <code>head</code> command, describe the columns of your
<code>.gff</code> file.
</li>
</ol>
</div>
<ol start="2" style="list-style-type: decimal">
<li>Use the following code to create a summary table of your functional annotations:</li>
</ol>
<pre><code>cut -f 7 your_sample_name.tsv | sort | uniq -c | sort -nr &gt; gene_counts.txt</code></pre>
<p>Open the <code>gene_counts.txt</code> file and answer the following questions:</p>
<div class="rmdquestion">
<ol start="5" style="list-style-type: decimal">
<li>
<p>
What is the most abundant gene prediction?
</p>
</li>
<li>
<p>
What is the most abundant gene prediction that is not a
hypothetical/predicted protein?
</p>
</li>
<li>
<p>
What are the most common five gene annotations in your
genome?
</p>
</li>
</ol>
</div>
<ol start="3" style="list-style-type: decimal">
<li>Finally, select three random <code>hypothetical proteins</code> from your <code>faa</code> file. Run them in InterPro Scan (<a href="https://www.ebi.ac.uk/interpro/search/sequence/" class="uri">https://www.ebi.ac.uk/interpro/search/sequence/</a>).</li>
</ol>
<p>Answer the following:</p>
<div class="rmdquestion">
<ol start="8" style="list-style-type: decimal">
<li>
<p>
Did any of these proteins have an annotation? Add it
below.
</p>
</li>
<li>
<p>
Add the path of your <code>faa</code> file here:
</p>
</li>
</ol>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="hybrid-assembly-methods-for-short-and-long-reads.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/08-geneprediction.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
