<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Hybrid Assembly Methods for Short and Long Reads | The Genome Project: Excercise Guidebook</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Hybrid Assembly Methods for Short and Long Reads | The Genome Project: Excercise Guidebook" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Hybrid Assembly Methods for Short and Long Reads | The Genome Project: Excercise Guidebook" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Javier F. Tabima Restrepo" />


<meta name="date" content="2023-02-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="assembly-methods-for-short-read-sequencing.html"/>
<link rel="next" href="gene-prediction-methods.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="font-awesome.min.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The Genome Project: A Guidebook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> BIOL209 (The Genome Project) Excercise Guidebook</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#additional-information-about-the-professor"><i class="fa fa-check"></i>Additional information about the professor</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="unix-challenge.html"><a href="unix-challenge.html"><i class="fa fa-check"></i><b>2</b> UNIX Challenge</a></li>
<li class="chapter" data-level="3" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html"><i class="fa fa-check"></i><b>3</b> FASTQ files and understanding read qualities</a>
<ul>
<li class="chapter" data-level="3.1" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html#sample-distribution"><i class="fa fa-check"></i><b>3.1</b> Sample distribution</a></li>
<li class="chapter" data-level="3.2" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html#basic-information-and-creating-backups"><i class="fa fa-check"></i><b>3.2</b> Basic information and creating backups</a></li>
<li class="chapter" data-level="3.3" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html#quality-control"><i class="fa fa-check"></i><b>3.3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html#evaluating-qc-of-fastq-using-fastqc"><i class="fa fa-check"></i><b>3.3.1</b> Evaluating QC of FASTQ using FastQC</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html"><i class="fa fa-check"></i><b>4</b> Laboratory techniques for DNA extraction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#protocol-using-g20-qiagen-columns"><i class="fa fa-check"></i><b>4.1</b> Protocol using G/20 QIAGEN columns</a></li>
<li class="chapter" data-level="4.2" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#quantification-of-dna-using-qubit-and-nanodrop"><i class="fa fa-check"></i><b>4.2</b> Quantification of DNA using Qubit and Nanodrop</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#qubit-flurescence-quantification"><i class="fa fa-check"></i><b>4.2.1</b> Qubit flurescence quantification</a></li>
<li class="chapter" data-level="4.2.2" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#nanodrop-dna-quality"><i class="fa fa-check"></i><b>4.2.2</b> Nanodrop DNA quality</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#analytical-gel-for-dna-integrity-and-fragment-size"><i class="fa fa-check"></i><b>4.3</b> Analytical gel for DNA integrity and fragment size</a></li>
<li class="chapter" data-level="4.4" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#dna-extraction-report"><i class="fa fa-check"></i><b>4.4</b> DNA extraction report</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html"><i class="fa fa-check"></i><b>5</b> Improving Illumina Read Quality</a>
<ul>
<li class="chapter" data-level="5.1" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html#running-trimmomatic"><i class="fa fa-check"></i><b>5.1</b> Running Trimmomatic</a></li>
<li class="chapter" data-level="5.2" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html#evaluating-quality-of-processed-reads"><i class="fa fa-check"></i><b>5.2</b> Evaluating quality of processed reads</a></li>
<li class="chapter" data-level="5.3" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html#requirements-for-processing-your-reads"><i class="fa fa-check"></i><b>5.3</b> Requirements for processing your reads</a></li>
<li class="chapter" data-level="5.4" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html#what-to-turn-in"><i class="fa fa-check"></i><b>5.4</b> What to turn in?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="nanopore-sequencing.html"><a href="nanopore-sequencing.html"><i class="fa fa-check"></i><b>6</b> Nanopore Sequencing</a>
<ul>
<li class="chapter" data-level="6.1" data-path="nanopore-sequencing.html"><a href="nanopore-sequencing.html#summarizing-our-nanopore-sequencing"><i class="fa fa-check"></i><b>6.1</b> Summarizing our Nanopore sequencing</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html"><i class="fa fa-check"></i><b>7</b> Assembly Methods for Short Read Sequencing</a>
<ul>
<li class="chapter" data-level="7.1" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#comparing-between-k-mers-in-genome-assembly"><i class="fa fa-check"></i><b>7.1</b> Comparing between k-mers in genome assembly</a></li>
<li class="chapter" data-level="7.2" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#using-a-multi-k-mer-approach-to-avoid-manual-k-mer-selection"><i class="fa fa-check"></i><b>7.2</b> Using a multi-K-mer approach to avoid manual k-mer selection</a></li>
<li class="chapter" data-level="7.3" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#cleaning-the-assembly-and-detecting-contamination"><i class="fa fa-check"></i><b>7.3</b> Cleaning the assembly and detecting contamination</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#sketch-assay"><i class="fa fa-check"></i><b>7.3.1</b> <strong>Sketch</strong> assay</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#coverage-and-taxonomy-detection-assay"><i class="fa fa-check"></i><b>7.4</b> Coverage and Taxonomy detection assay</a></li>
<li class="chapter" data-level="7.5" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#extracting-the-scaffolds-of-interest"><i class="fa fa-check"></i><b>7.5</b> Extracting the scaffolds of interest</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="hybrid-assembly-methods-for-short-and-long-reads.html"><a href="hybrid-assembly-methods-for-short-and-long-reads.html"><i class="fa fa-check"></i><b>8</b> Hybrid Assembly Methods for Short and Long Reads</a>
<ul>
<li class="chapter" data-level="8.1" data-path="hybrid-assembly-methods-for-short-and-long-reads.html"><a href="hybrid-assembly-methods-for-short-and-long-reads.html#hybrid-assembly-using-unicycler"><i class="fa fa-check"></i><b>8.1</b> Hybrid assembly using <code>Unicycler</code></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html"><i class="fa fa-check"></i><b>9</b> Gene Prediction Methods</a>
<ul>
<li class="chapter" data-level="9.1" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html#installing-prokka"><i class="fa fa-check"></i><b>9.1</b> Installing prokka</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html#installing-the-prokka-environment-and-program"><i class="fa fa-check"></i><b>9.1.1</b> Installing the prokka environment and program</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html#annotating-the-genomes-using-prokka"><i class="fa fa-check"></i><b>9.2</b> Annotating the genomes using prokka</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html#summarizing-the-results-from-prokka"><i class="fa fa-check"></i><b>9.2.1</b> Summarizing the results from prokka</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Genome Project: Excercise Guidebook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="hybrid-assembly-methods-for-short-and-long-reads" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Hybrid Assembly Methods for Short and Long Reads<a href="hybrid-assembly-methods-for-short-and-long-reads.html#hybrid-assembly-methods-for-short-and-long-reads" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="objectives">
<p>
<em>Objectives</em>
</p>
<ol style="list-style-type: decimal">
<li>
</li>
<li>
</li>
<li>
</li>
</ol>
</div>
<p>In the modern era of cheaper sequencing and easily accessible genome sequencers, we can also include long reads in our assembly. The long reads will provide the assemblers with longer regions in which the different fragments can anchor and connect, creating, in theory, longer scaffolds and a more complete assembly. Conversely, these long reads usually have lower quality than short reads and they are less abundant, so using them by themselves may lead to an assembly with a larger percentage or errors than, for example, an assembly with only short reads.</p>
<p>So, the best of both worlds can be combined into what we call <strong>hybrid assemblies</strong>. Hybrid assemblies use both short and long reads to produce a high quality assembly (compred to an assembly of long reads only) with longer and fewer scaffolds than assemblies done with only short reads.</p>
<p>To perform hybrid genome assemblies, we need two main files:</p>
<ol style="list-style-type: decimal">
<li>Our Illumina reads (Choose between raw reads or cleaned reads)</li>
<li>Our Nanopore reads (Located at the <code>/Smaug_SSD/BIOL209/nanopore_raw_data</code> folder)</li>
</ol>
<div id="hybrid-assembly-using-unicycler" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Hybrid assembly using <code>Unicycler</code><a href="hybrid-assembly-methods-for-short-and-long-reads.html#hybrid-assembly-using-unicycler" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The program <code>unicycler</code> is combination of programs (also called a <strong>pipeline</strong>) to perform a hybrid assembly. Usually, a manual hybrid assembly (i.e. using each program separately) entails the following steps:</p>
<ol style="list-style-type: decimal">
<li><code>Spades</code> to make the short reads assemblies</li>
<li><code>miniasm</code> and <code>racon</code> to bridge between the scaffolds from the short read assemblies</li>
<li><code>bwa</code> or <code>bowtie</code> to map the reads to the new elongated scaffolds for error control</li>
<li><code>pilon</code> to correct the errors detected by the previous read mapping step</li>
</ol>
<p>However, for prokariotic organisms as the ones we are trying to assemble, our <code>Unicycler</code> program will do everything mentioned above and more, so hopefully we will have the best assambly possible using the highest amount of data.</p>
<p>To run <code>Unicycler</code> with the long read and short read datasets, modify the following command with your <code>FASTQ</code> reads:</p>
<pre><code>unicycler -1 paired_ends_R1.fastq.gz -2 paired_ends_R2.fastq.gz -l Nanopore_reads.fastq.gz -o long_read_assembly</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Let Unicycler run. Answer the following questions:</li>
</ol>
<div class="rmdquestion">
<ol style="list-style-type: decimal">
<li>
Fill the following table using the information from the hybrid
assembly and your previous best short read assembly (Hint: use the
output from <code>stats.sh</code>):
</li>
</ol>
<table>
<thead>
<tr class="header">
<th>
Source
</th>
<th>
Number of scaffolds
</th>
<th>
%GC
</th>
<th>
N50
</th>
<th>
L90
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>
Short read assembly
</td>
<td>
</td>
<td>
</td>
<td>
</td>
<td>
</td>
</tr>
<tr class="even">
<td>
Hybrid assembly
</td>
<td>
</td>
<td>
</td>
<td>
</td>
<td>
</td>
</tr>
</tbody>
</table>
</div>
<ol start="3" style="list-style-type: decimal">
<li>Use a <strong>Sketch</strong> assay to determine the taxonomic markup of your long read assembly and answer the following questions for the <strong>hybrid assembly</strong>:</li>
</ol>
<div class="rmdquestion">
<ol start="2" style="list-style-type: decimal">
<li>
<p>
What are the <strong>GENUS</strong> with higher WKID
scores?
</p>
</li>
<li>
<p>
What is the WKID score of your assembly when compared to
<em>Synechococcus</em>?
</p>
</li>
<li>
<p>
Summarize you results and what they mean
</p>
</li>
</ol>
</div>
<ol start="4" style="list-style-type: decimal">
<li>Use <code>blobtools</code> to identify the taxonomic markup of your long read assembly and answer the following questions (Only map your short reads to the new assembly. Remember to use the <code>bwa index long_read_assembly.fasta</code> before you map the reads using <code>bwa mem</code>)</li>
</ol>
<div class="rmdquestion">
<ol start="5" style="list-style-type: decimal">
<li>
<p>
Include the image here:
</p>
</li>
<li>
<p>
What is the taxonomical unit with the highest percentage of
scaffolds in our assembly?
</p>
</li>
<li>
<p>
What is the percentage of scaffolds with assignment to
<em>Synechococcus</em>?
</p>
</li>
</ol>
</div>
<ol start="5" style="list-style-type: decimal">
<li><p>Finally, extract all the contigs/scaffolds with Cyanobacteria ID. Save these scaffolds in a file called <code>final_cyano_assembly.fasta</code></p></li>
<li><p>Map the short reads to the <code>final_cyano_assembly.fasta</code> using <code>BWA</code>. <strong>NOTE: Please remember to do the <code>bwa index final_cyano_assembly.fasta</code> code</strong></p></li>
<li><p>Use the <code>pileup.sh</code> program to summarize the coverage results. Save the number of mapped reads and the percentage of reads mapped.</p></li>
<li><p>We will do a similar thing using long reads. However, <code>BWA</code> is only used for short reads. The creators of <code>BWA</code> made a long read mapper called <code>minimap2</code>. <code>minimap2</code> maps the reads and creates a <code>BAM</code> file like in <code>BWA</code> but using the long reads. Lets use it to see our coverage with the long read data:</p></li>
</ol>
<pre><code>minimap2 -a final_cyano_assembly.fasta Nanopore_reads.fastq.gz |samtools view -S -b | samtools sort -O bam -o long_read_mapped.bam</code></pre>
<p>Where the <code>-a</code> means create a <code>SAM</code> file that will be converted into a <code>BAM</code> file.</p>
<div class="rmdquestion">
<ol start="8" style="list-style-type: decimal">
<li>
Use the <code>pileup.sh</code> program to summarize the coverage
results and fill the following table.
</li>
</ol>
<table>
<thead>
<tr class="header">
<th>
Dataset
</th>
<th>
Number of reads
</th>
<th>
Percentage of mapped reads
</th>
<th>
Average coverage
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>
Short read assembly
</td>
<td>
</td>
<td>
</td>
<td>
</td>
</tr>
<tr class="even">
<td>
Hybrid assembly
</td>
<td>
</td>
<td>
</td>
<td>
</td>
</tr>
</tbody>
</table>
<ol start="9" style="list-style-type: decimal">
<li>
Did your long reads help improve the assembly? Justify your
answer.
</li>
</ol>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="assembly-methods-for-short-read-sequencing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gene-prediction-methods.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/07-hybridassembly.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
