<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Improving Illumina Read Quality | The Genome Project: Excercise Guidebook</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.28 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Improving Illumina Read Quality | The Genome Project: Excercise Guidebook" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Improving Illumina Read Quality | The Genome Project: Excercise Guidebook" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Javier F. Tabima Restrepo" />


<meta name="date" content="2023-01-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="laboratory-techniques-for-dna-extraction.html"/>
<link rel="next" href="nanopore-sequencing.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="font-awesome.min.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The Genome Project: A Guidebook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> BIOL209 (The Genome Project) Excercise Guidebook</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#additional-information-about-the-professor"><i class="fa fa-check"></i>Additional information about the professor</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="unix-challenge.html"><a href="unix-challenge.html"><i class="fa fa-check"></i><b>2</b> UNIX Challenge</a></li>
<li class="chapter" data-level="3" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html"><i class="fa fa-check"></i><b>3</b> FASTQ files and understanding read qualities</a>
<ul>
<li class="chapter" data-level="3.1" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html#sample-distribution"><i class="fa fa-check"></i><b>3.1</b> Sample distribution</a></li>
<li class="chapter" data-level="3.2" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html#basic-information-and-creating-backups"><i class="fa fa-check"></i><b>3.2</b> Basic information and creating backups</a></li>
<li class="chapter" data-level="3.3" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html#quality-control"><i class="fa fa-check"></i><b>3.3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="fastq-files-and-understanding-read-qualities.html"><a href="fastq-files-and-understanding-read-qualities.html#evaluating-qc-of-fastq-using-fastqc"><i class="fa fa-check"></i><b>3.3.1</b> Evaluating QC of FASTQ using FastQC</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html"><i class="fa fa-check"></i><b>4</b> Laboratory techniques for DNA extraction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#protocol-using-g20-qiagen-columns"><i class="fa fa-check"></i><b>4.1</b> Protocol using G/20 QIAGEN columns</a></li>
<li class="chapter" data-level="4.2" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#quantification-of-dna-using-qubit-and-nanodrop"><i class="fa fa-check"></i><b>4.2</b> Quantification of DNA using Qubit and Nanodrop</a></li>
<li class="chapter" data-level="4.3" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#analytical-gel-for-dna-integrity-and-size"><i class="fa fa-check"></i><b>4.3</b> Analytical gel for DNA integrity and size</a></li>
<li class="chapter" data-level="4.4" data-path="laboratory-techniques-for-dna-extraction.html"><a href="laboratory-techniques-for-dna-extraction.html#dna-extraction-report"><i class="fa fa-check"></i><b>4.4</b> DNA extraction report</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html"><i class="fa fa-check"></i><b>5</b> Improving Illumina Read Quality</a>
<ul>
<li class="chapter" data-level="5.1" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html#running-trimmomatic"><i class="fa fa-check"></i><b>5.1</b> Running Trimmomatic</a></li>
<li class="chapter" data-level="5.2" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html#evaluating-quality-of-processed-reads"><i class="fa fa-check"></i><b>5.2</b> Evaluating quality of processed reads</a></li>
<li class="chapter" data-level="5.3" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html#requirements-for-processing-your-reads"><i class="fa fa-check"></i><b>5.3</b> Requirements for processing your reads</a></li>
<li class="chapter" data-level="5.4" data-path="improving-illumina-read-quality.html"><a href="improving-illumina-read-quality.html#what-to-turn-in"><i class="fa fa-check"></i><b>5.4</b> What to turn in?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="nanopore-sequencing.html"><a href="nanopore-sequencing.html"><i class="fa fa-check"></i><b>6</b> Nanopore Sequencing</a>
<ul>
<li class="chapter" data-level="6.1" data-path="nanopore-sequencing.html"><a href="nanopore-sequencing.html#summarizing-our-nanopore-sequencing"><i class="fa fa-check"></i><b>6.1</b> Summarizing our Nanopore sequencing</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html"><i class="fa fa-check"></i><b>7</b> Assembly Methods for Short Read Sequencing</a>
<ul>
<li class="chapter" data-level="7.1" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#comparing-between-k-mers-in-genome-assembly"><i class="fa fa-check"></i><b>7.1</b> Comparing between k-mers in genome assembly</a></li>
<li class="chapter" data-level="7.2" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#using-a-multi-k-mer-approach-to-avoid-manual-k-mer-selection"><i class="fa fa-check"></i><b>7.2</b> Using a multi-K-mer approach to avoid manual k-mer selection</a></li>
<li class="chapter" data-level="7.3" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#cleaning-the-assembly-and-detecting-contamination"><i class="fa fa-check"></i><b>7.3</b> Cleaning the assembly and detecting contamination</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#sketch-assay"><i class="fa fa-check"></i><b>7.3.1</b> <strong>Sketch</strong> assay</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#coverage-and-taxonomy-detection-assay"><i class="fa fa-check"></i><b>7.4</b> Coverage and Taxonomy detection assay</a></li>
<li class="chapter" data-level="7.5" data-path="assembly-methods-for-short-read-sequencing.html"><a href="assembly-methods-for-short-read-sequencing.html#extracting-the-scaffolds-of-interest"><i class="fa fa-check"></i><b>7.5</b> Extracting the scaffolds of interest</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="hybrid-assembly-methods-for-short-and-long-reads.html"><a href="hybrid-assembly-methods-for-short-and-long-reads.html"><i class="fa fa-check"></i><b>8</b> Hybrid Assembly Methods for Short and Long Reads</a>
<ul>
<li class="chapter" data-level="8.1" data-path="hybrid-assembly-methods-for-short-and-long-reads.html"><a href="hybrid-assembly-methods-for-short-and-long-reads.html#hybrid-assembly-using-unicycler"><i class="fa fa-check"></i><b>8.1</b> Hybrid assembly using <code>Unicycler</code></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html"><i class="fa fa-check"></i><b>9</b> Gene Prediction Methods</a>
<ul>
<li class="chapter" data-level="9.1" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html#installing-prokka"><i class="fa fa-check"></i><b>9.1</b> Installing prokka</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html#installing-the-prokka-environment-and-program"><i class="fa fa-check"></i><b>9.1.1</b> Installing the prokka environment and program</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html#annotating-the-genomes-using-prokka"><i class="fa fa-check"></i><b>9.2</b> Annotating the genomes using prokka</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="gene-prediction-methods.html"><a href="gene-prediction-methods.html#summarizing-the-results-from-prokka"><i class="fa fa-check"></i><b>9.2.1</b> Summarizing the results from prokka</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Genome Project: Excercise Guidebook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="improving-illumina-read-quality" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Improving Illumina Read Quality<a href="improving-illumina-read-quality.html#improving-illumina-read-quality" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="objectives">
<p>
<strong>Objectives and goal</strong>
</p>
<p>
The goal of this challenge is to properly process the Illumina data
in preparation for genome assembly. You work in your lab groups and use
forward and reverse reads to completely remove and/or trim parts of the
reads that are low quality and problematic while still maintaining the
highest number of sequence data as possible.
</p>
<p>
How you decide to process, trim, and filter your reads is up to you.
I’d recommend you use Trimmomatic, but if you want to write your own
custom scripts feel welcome to do so. Be sure to consider the problems
that came to light in the quality analysis from Lab 2. The key to this
challenge is in how you use the various settings of available programs.
Take a careful look at the manual pages for these tools to take
advantage of all of their capabilities.
</p>
</div>
<hr />
<div id="running-trimmomatic" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Running Trimmomatic<a href="improving-illumina-read-quality.html#running-trimmomatic" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>I recommend you all check the slides of the <strong>Short read sequencing</strong> class as they have visual instructions on how to use <code>Trimmomatic</code>. Check the options!!</p>
<p>To run <code>trimmomatic</code> in the cluster, modify the following template:</p>
<pre><code>java -jar /Smaug_SSD/bin/Trimmomatic-0.39/trimmomatic-0.39.jar PE -T 1 R1.fastq.gz R2.fastq.gz Processed_R1.fastq.gz Processed_R1.Unpaired.fastq.gz Processed_R2.fastq.gz Processed_R2.Unpaired.fastq.gz OPTIONS_FOR_TRIMMING</code></pre>
<p>Example for Illumina clip:</p>
<pre><code>java -jar /Smaug_SSD/bin/Trimmomatic-0.39/trimmomatic-0.39.jar PE -threads 1 NBP03_S165_R1_001.fastq.gz NBP03_S165_R2_001.fastq.gz IlluminaTrim_R1.fastq.gz IlluminaTrim_R1.Unpaired.fastq.gz lluminaTrim_R2.fastq.gz lluminaTrim_R2.Unpaired.fastq.gz ILLUMINACLIP:/Smaug_SSD/bin/Trimmomatic-0.39/adapters/TruSeq3-PE.fa:2:30:10:2:True</code></pre>
<p>Check the <code>trimmomatic</code> web page for more info: <a href="https://github.com/usadellab/Trimmomatic" class="uri">https://github.com/usadellab/Trimmomatic</a></p>
<p>I also expect you ask about what is an unpaired read and why are they part of the output.</p>
<hr />
</div>
<div id="evaluating-quality-of-processed-reads" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Evaluating quality of processed reads<a href="improving-illumina-read-quality.html#evaluating-quality-of-processed-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The team with the largest number of bases mapped to the reference genome of <strong>Synechococcus</strong> sequenced by the 2019 BIOL209 course will ‘win’ the competition. The srefernece sequence can be found at <code>/Smaug_SSD/BIOL209/Reference_Syn/GCA_013133755.1_ASM1313375v1_genomic.fasta</code>. We will evaluate the results on next week’s lab.</p>
<p>However, if you want to evaluate the quality and quantity of your resulting fastq files, you must run a read alignment against the BIOL209 reference genome. I suggest that you test out your pipeline then run the mapping program to evaluate you results. Try different settings and repeat until you’re satisfied. Please be mindful of the space you are taking up in your folders, as the intermediate fastq files you generate for multiple tests and steps can add up. Please consider deleting any .fastq data along the way that you no longer need. After the exercise is complete, I’ll ask you delete any extraneous fastq files as well.</p>
<p>Below is the command that you must use to generate the mapping output and evaluate the quantity of your resulting reads. The program is called <code>BBMap</code> (<a href="https://jgi.doe.gov/data-and-tools/bbtools/" class="uri">https://jgi.doe.gov/data-and-tools/bbtools/</a>). We will learn more about read mapping in the upcoming weeks, but for this lab please use the following script:</p>
<blockquote>
<p>Note: Be sure to copy the reference genome to a folder in your own account directory for bbsplit to work properly as it will generate a folder to store reference info.</p>
</blockquote>
<pre><code>/Smaug_SSD/bin/bbmap/bbmap.sh in=Processed_R1.fastq.gz in2=Processed_R2.fastq.gz ref=GCA_013133755.1_ASM1313375v1_genomic.fasta minid=0.90 threads=16 statsfile=stats_out.txt
</code></pre>
<p>Check the <code>stats_out.txt</code> file as it will include your results as such:</p>
<pre><code>Read 1 data:        pct reads   num reads   pct bases      num bases
mapped:               0.4267%        8726     0.4267%        2181500</code></pre>
<hr />
</div>
<div id="requirements-for-processing-your-reads" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Requirements for processing your reads<a href="improving-illumina-read-quality.html#requirements-for-processing-your-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To make fair comparisons, the processing of your reads must meet these requirements:</p>
<ul>
<li>All processed reads must have a minimum length of 36 bp.</li>
<li>Use a minimum read Phred quality score cutoff of 20.</li>
</ul>
<hr />
</div>
<div id="what-to-turn-in" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> What to turn in?<a href="improving-illumina-read-quality.html#what-to-turn-in" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Grading will be based on you providing:</p>
<ul>
<li>The commands for your sequence processing pipeline</li>
<li>The updated HTML of FASTQC before and after processing the reads</li>
<li>The location of your processed reads within <code>Smaug</code></li>
<li>The output of BBmap’s <code>statsfile</code></li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="laboratory-techniques-for-dna-extraction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="nanopore-sequencing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/04-illumina.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
